### PR TITLE
NEW Use supported modules repo for branch-based logic

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/rhino/issues/17

### CODE REVIEW

File: tests/BranchesTest.php [hunk 1]
```diff
@@ -10,26 +10,36 @@ class BranchesTest extends TestCase
...
-            $tagsJson
-        );
-        if (!$expectException) {
-            $this->assertSame($expected, $actual);
+        try {
+            if ($composerJson) {
+                file_put_contents('__composer.json', $composerJson);
+            }
+            file_put_contents('__branches.json', $branchesJson);
+            file_put_contents('__tags.json', $tagsJson);
```
Comment (ChloeSartorelli) [inline tests/BranchesTest.php:27]: Putting the files there means we don't have a separate code path for tests than we do for normal execution. This is the same as how we handle composer.json data in the generate-matrix tests.

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    },
+    "require": {
+        "silverstripe/supported-modules": "dev-pulls/main/combine-all-versions"
+    },
+    "repositories": {
+        "silverstripe/supported-modules": {
+            "type": "vcs",
+            "url": "git@github.com:creative-commoners/supported-modules.git"
+        }
+    }
```
Comment (emteknetnz) [inline composer.json]:
```suggestion
    "require": {
        "silverstripe/supported-modules": "dev-main"
    },
    "require-dev": {
        "phpunit/phpunit": "^9.6"
    }
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (ChloeSartorelli) [reply composer.json]: Yup, that'll be done once the PR for that fork is merged, so we can easily rerun CI here against that PR to see it go green if changes are needed over there.

Comment (ChloeSartorelli) [reply composer.json]: Done now that the fork PR is merged and the repo is in packagist.

**Review by emteknetnz — APPROVED** [review]

**Comment by emteknetnz** [comment]
@GuySartorelli Remember to create a new minor branch and manually tag

**Pull request merged**